File: Range Sum Query - Immutable.js
var NumArray = function(nums) {
        this.nums = nums
        for(let i=1; i<nums.length; i++) this.nums[i]=this.nums[i] + this.nums[i-1];
};

NumArray.prototype.sumRange = function(left, right) {
    if(left === 0) return this.nums[right];
    else return this.nums[right] - this.nums[left-1];
};

File: Range Sum Query - Immutable.py
class NumArray:

    def __init__(self, nums: List[int]):
        self.nums = nums
        for i in range(1, len(nums)):
            self.nums[i] += self.nums[i-1]
        self.nums.append(0)
    def sumRange(self, left: int, right: int) -> int:
        return self.nums[right] - self.nums[left-1]
